addons/hr_holidays/tests/test_negative.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import datetime
+from freezegun import freeze_time
+
+from odoo.tests.common import tagged
+from odoo.addons.hr_holidays.tests.common import TestHrHolidaysCommon
+from odoo.models import ValidationError
+
+
+@tagged('negative_time_off')
+class TestNegative(TestHrHolidaysCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.leave_type = cls.env['hr.leave.type'].create({
+            'name': 'Limited with negative',
+            'leave_validation_type': 'no_validation',
+            'requires_allocation': 'yes',
+            'company_id': cls.company.id,
+            'allows_negative': True,
+            'max_allowed_negative': 5,
+        })
+
+        cls.allocation_2022 = cls.env['hr.leave.allocation'].create({
+            'employee_id': cls.employee_emp_id,
+            'holiday_status_id': cls.leave_type.id,
+            'date_from': '2022-01-01',
+            'date_to': '2022-12-31',
+            'number_of_days': 1,
+        })
+
+        cls.allocation_2023 = cls.env['hr.leave.allocation'].create({
+            'employee_id': cls.employee_emp_id,
+            'holiday_status_id': cls.leave_type.id,
+            'date_from': '2023-01-01',
+            'number_of_days': 5,
+        })
+
+    def test_negative_time_off(self):
+        with freeze_time('2022-10-02'):
+            self.env['hr.leave'].with_user(self.user_employee_id).create({
+                'name': 'first',
```

Comment:
@Bertrand2, maybe more explanatory naming? Like 'First leave with x days'

Also, on top, a bit of math commentary would be nice too, like:

```available - 
   20 allocation days
   5 negative cap
  ----------------
  25 alltogehter
```

Later on, you can also add taken days